### PR TITLE
upgrade to datadog go client

### PR DESCRIPTION
**Pull request by geseq**

### CODE REVIEW

**Comment by geseq** [comment]
this is the initial PR with datadog's own defaults. not really changing anything other than using datadog as a client here  

i will follow this up with separate PRs for configuring specific aspects such as aggregation config.

**Comment by geseq** [comment]
also added a new function to allow custom configured client. this will basically simplify the passing of datadog specific options with minimal changes and is BC

File: stats/dogstatsd.go [hunk 1]
```diff
@@ -6,76 +6,32 @@ import (
 	"log"
 	"time"
 
-	"github.com/PagerDuty/godspeed"
+	ddsd "github.com/DataDog/datadog-go/v5/statsd"
```
Comment (dmotylev) [inline stats/dogstatsd.go:9]: why not `datadog`?

File: stats/dogstatsd.go [hunk 2]
```diff
@@ -155,37 +114,31 @@ func (d *dogstatsd) Error(err error, tags ...string) {
 }
 
 func (d *dogstatsd) Count(stat string, count float64, tags ...string) {
-	d.send <- &statsd{&statsdDatum{stat, "c", count, 1}, nil, tags}
+	if err := d.a.Count(stat, int64(count), tags, 1); err != nil {
+		log.Printf("Unable to send stats data: %s", err)
+	}
```
Comment (dmotylev) [inline stats/dogstatsd.go:119]: This will create some colossal storm of noise in case of an error, won't it?

**Review by dmotylev — COMMENTED** [review]
Yes, looks good; I'm not sure about error handling though. Shall we be smart with it and use some rate limiter?

File: stats/dogstatsd.go (hunk 2 above)
Comment (geseq) [reply stats/dogstatsd.go:119]: It was in the old implementation as well so I thought probably not, but might be best to double check. I’ll remove it if I find it to be noisy.

File: stats/dogstatsd.go (hunk 1 above)
Comment (geseq) [reply stats/dogstatsd.go]: i'm so bad at naming :laughing:

**Review by dmotylev — APPROVED** [review]

**Comment by geseq** [comment]
tested on jmap stg. works fine.

**Pull request merged**